### ISSUE TITLE
SerializerBase.Serialize produces truncated string

### ISSUE THREAD

**Issue by mcshaz** [ical-org/ical.net#157]
[SerializerBase](https://github.com/rianjs/ical.net/blob/master/ical.NET/Serialization/iCalendar/Serializers/SerializerBase.cs) has the code:

``` C#
public void Serialize(object obj, Stream stream, Encoding encoding)
{
        // NOTE: we don't use a 'using' statement here because
        // we don't want the stream to be closed by this serialization.
       // Fixes bug #3177278 - Serialize closes stream
       var sw = new StreamWriter(stream, encoding);
       // Push the current object onto the serialization stack
       SerializationContext.Push(obj);
       // Push the current encoding on the stack
       var encodingStack = GetService<IEncodingStack>();
       encodingStack.Push(encoding);
       sw.Write(SerializeToString(obj));
       // Pop the current encoding off the serialization stack
       encodingStack.Pop();
       // Pop the current object off the serialization stack
      SerializationContext.Pop();
}
```

This is resulting in the stream having a truncated string written to it if that string is bigger than the default buffer size. 

the 2 options are:
1. call sw.Flush() after sw.Write();
2. wrap the StreamWriter in a using block but use the overload StreamWriter(Stream stream, Encoding encoding, int bufferSize, bool leaveOpen) & set leaveOpen to true

**Comment by rianjs** [comment]
ae6f95dcc1c5d36d76bdb123ffd9d91f9d2369b9

**Comment by rianjs** [comment]
Naturally .NET 4.0 doesn't have a `leaveOpen` constructor parameter for streams.

I really should just leave .NET 4.0 behind. Supporting it is way more terrible than it should be.

**Comment by rianjs** [comment]
Fix published in Nuget package 2.2.15
